hooks/configure

```diff
@@ -14,74 +14,108 @@
-    esac
-
-}
-
-switch_service() {
-	case "$1" in
-		false)
-			# When the unit is already enabled but not active a call with
-			# --now does start the unit so we have to check for that  case
-			# and explicitly start the unit.
-			if [ "$(systemctl is-enabled "$2.service")" = "disabled" ]; then
-				systemctl enable "$2.service"
-			fi
-			if [ "$(systemctl is-active "$2.service")" = "inactive" ]; then
-				systemctl start "$2.service"
-			fi
-			;;
-		true)
-			# A simple `systemctl disable --now ssh.service` doesn't work
-			# and fails with an error message. Because of that we're going
-			# in two steps here and disable first and then stopping the
-			# service unit.
-			systemctl disable "$2.service"
-			systemctl stop "$2.service"
-			;;
-		*)
-			echo "ERROR: Invalid value '$1' provided for option service.$2.disable"
-			exit 1
-	esac
-}
-
-for service in $SERVICES; do
-    value=$(snapctl get service.$service.disable)
-    if [ -n "$value" ]; then
-        switch_service "$value" "$service"
-    fi
-done
-
-powerkey=$(snapctl get system.power-key-action)
-if [ -n "$powerkey" ]; then
-    switch_handle_power_key "$powerkey"
-fi
+SERVICES=("ssh",)
+
+
+def systemctl(*args, check=False):
+    res = subprocess.run(["systemctl"]+list(args), stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    if check and res.returncode != 0:
+        raise subprocess.CalledProcessError()
+    return res
+
+
+def core_support_available():
+    """Return true if core_support interface is available
+
+    The check is done indirectly via checking if systemctl can be used.
+    """
+    return systemctl("--version").returncode == 0
+
+
+def switch_handle_power_key(action):
+    valid_actions = ("ignore","poweroff","reboot","halt","kexec","suspend","hibernate","hybrid-sleep","lock")
+    login_conf_d=os.getenv("TEST_LOGIN_CONF_D", default="/etc/systemd/logind.conf.d")
+    conf=os.path.join(login_conf_d, "00-snap-core.conf")
+
+    if not os.path.isdir(login_conf_d):
+        os.makedirs(login_conf_d)
+    if not action in valid_actions:
+        raise Exception("invalid action '{}' supplied for system.power-key-action option".format(action))
+    with open(conf+".tmp", "w") as fp:
+        fp.write("[Login]\nHandlePowerKey={}\n".format(action))
+    os.rename(conf+".tmp", conf)
+
+
+def switch_service(service, enable):
+    srv = "{}.service".format(service)
+    if enable == "true":
+ 	# A simple `systemctl disable --now ssh.service` doesn't work
+ 	# and fails with an error message. Because of that we're going
+ 	# in two steps here and disable first and then stopping the
+ 	# service unit.
+        systemctl("disable", srv)
+        systemctl("stop", srv)
+    elif enable == "false":
+ 	# When the unit is already enabled but not active a call with
+ 	# --now does start the unit so we have to check for that  case
+ 	# and explicitly start the unit.
+        if systemctl("is-enabled", srv).stdout.strip() == "disabled":
```

Comment:
did you know is-enabled and is-active also set the return value? they have a --quiet option to not print anything if you want to go that route instead